test/test_automatic_segmentation.py

```diff
@@ -0,0 +1,153 @@
+import unittest
+
+import numpy as np
+from skimage.draw import disk
+from skimage.measure import label as connected_components
+
+import micro_sam.util as util
+
+
+class TestAutomaticSegmentation(unittest.TestCase):
+    model_type = "vit_t" if util.VIT_T_SUPPORT else "vit_b"
+    model_type_ais = "vit_t_lm" if util.VIT_T_SUPPORT else "vit_b_lm"
+    tile_shape = (512, 512)
+    halo = (96, 96)
+
+    # create an input 2d image with three objects
+    @staticmethod
+    def _get_2d_inputs(shape=(256, 256)):
+        mask = np.zeros(shape, dtype="uint8")
+
+        def write_object(center, radius):
+            circle = disk(center, radius, shape=shape)
+            mask[circle] = 1
+
+        center = tuple(sh // 4 for sh in shape)
+        write_object(center, radius=29)
+
+        center = tuple(sh // 2 for sh in shape)
+        write_object(center, radius=33)
+
+        center = tuple(3 * sh // 4 for sh in shape)
+        write_object(center, radius=35)
+
+        image = mask * 255
+        mask = connected_components(mask)
+        return mask, image
+
+    # create an input 2d image with three objects and stack 8 of them together
+    @classmethod
+    def _get_3d_inputs(cls, shape=(8, 256, 256)):
+        mask, image = cls._get_2d_inputs(shape[-2:])
+
+        # Create volumes by stacking the input image and respective mask.
+        volume = np.stack([image] * 8)
+        labels = np.stack([mask] * 8)
```

Comment:
You should use `shape[0]` rather than hard-coding this.

Reply:
Fixed. Thanks for spotting!